packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -122,18 +121,18 @@ library LibVerifying {
                 // It's not possible to verify this block if either the
                 // transition is contested and awaiting higher-tier proof or if
                 // the transition is still within its cooldown period.
+                uint16 tier = ts.tier;
+
                 if (ts.contester != address(0)) {
                     break;
                 } else {
-                    if (tierProvider == ITierProvider(address(0))) {
```

Comment:
we cannot assume the tier provider for block N is also the same one as block N+1 in the future.